docs/core/bluetooth/bluetooth_fetching_data.md

```diff
@@ -0,0 +1,398 @@
+        entry.entry_id
+    ] = ActiveBluetoothProcessorCoordinator(
+        hass,
+        _LOGGER,
+        address=address,
+        mode=BluetoothScanningMode.PASSIVE,
+        update_method=data.update,
+        needs_poll_method=_needs_poll,
+        poll_method=_async_poll,
+        # We will take advertisements from non-connectable devices
+        # since we will trade the BLEDevice for a connectable one
+        # if we need to poll it
+        connectable=False,
+    )
+    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+    entry.async_on_unload(
+        # only start after all platforms have had a chance to subscribe
+        coordinator.async_start()
+    )
+    return True
+```
+
+## BluetoothCoordinator
+
+The `ActiveBluetoothCoordinator` and `PassiveBluetoothCoordinator` coordinators function similar
+to `DataUpdateCoordinators` except they are driven by incoming advertisement data instead of polling.
+
+### PassiveBluetoothCoordinator
+
+Below is an example of a `PassiveBluetoothDataUpdateCoordinator`. Incoming
+data is received via `_async_handle_bluetooth_event` and processed by the integration's
+library.
+
+```python
+import logging
+from typing import TYPE_CHECKING
+
+from homeassistant.components import bluetooth
+from homeassistant.components.bluetooth.active_update_coordinator import (
+    PassiveBluetoothDataUpdateCoordinator,
+)
+from homeassistant.core import CoreState, HomeAssistant, callback
+
+if TYPE_CHECKING:
+    from bleak.backends.device import BLEDevice
+
+
+class ExamplePassiveBluetoothDataUpdateCoordinator(
+    PassiveBluetoothDataUpdateCoordinator[None]
+):
+    """Class to manage fetching example data."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        logger: logging.Logger,
+        ble_device: BLEDevice,
+        device: YourLibDevice,
+    ) -> None:
+        """Initialize example data coordinator."""
+        super().__init__(
+            hass=hass,
+            logger=logger,
+            address=ble_device.address,
+            mode=bluetooth.BluetoothScanningMode.ACTIVE,
+            connectable=False,
+        )
+        self.device = device
+
+    @callback
+    def _async_handle_unavailable(
+        self, service_info: bluetooth.BluetoothServiceInfoBleak
+    ) -> None:
+        """Handle the device going unavailable."""
+
+    @callback
+    def _async_handle_bluetooth_event(
+        self,
+        service_info: bluetooth.BluetoothServiceInfoBleak,
+        change: bluetooth.BluetoothChange,
+    ) -> None:
+        """Handle a Bluetooth event."""
+        # Your device should process incoming advertisement data
+
+```
+
+### ActiveBluetoothCoordinator
+
+Below is an example of an `ActiveBluetoothDataUpdateCoordinator`. Incoming
+data is received via `_async_handle_bluetooth_event` and processed by the integration's
+library.
+
+The `_needs_poll` function is called each time there is the Bluetooth advertisement changes to
```

Comment:
This sentence doesn't sound right grammar wise.

Reply:
I read it a couple times and maybe its just my brain filling in the missing pieces subconsciously but it seemed fine to me so I rephrased it in https://github.com/home-assistant/developers.home-assistant/pull/1640 . Hopefully it makes more sense.